subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java

```diff
@@ -144,6 +144,7 @@ public void setToolChain(JavaToolChain toolChain) {
     protected void compile(@SuppressWarnings("deprecation") org.gradle.api.tasks.incremental.IncrementalTaskInputs inputs) {
         DeprecationLogger.deprecate("Extending the JavaCompile task")
             .withAdvice("Configure the task instead.")
+            .withDslReferenceForType(JavaCompile.class)
```

Comment:
I think this is another behavior changing deprecation that needs an upgrade section added for it.  I would defer for now

Reply:
Leaving undocumented for now.